### PR TITLE
K8SPXC-260 SecuritContext text update for Openshift

### PR DESCRIPTION
**Pull request by cap1984**
[K8SPXC-260](https://jira.percona.com/browse/K8SPXC-260)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | failed |
| big-data | failed |
| init-deploy | failed |
| recreate | passed |
| limits | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| monitoring | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| scaling-proxysql | passed |
| self-healing-advanced | failed |
| demand-backup | failed |
| affinity | passed |
| upgrade | passed |
| one-pod | passed |
| operator-self-healing | failed |
| scheduled-backup | failed |
| upgrade-consistency | passed |
| security-context | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| big-data | passed |
| scaling | passed |
| recreate | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| demand-backup | failed |
| upgrade-consistency | passed |
| security-context | passed |
| init-deploy | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing | passed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | failed |
| init-deploy | failed |
| storage | failed |
| recreate | passed |
| scaling | failed |
| limits | passed |
| monitoring | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| restore-to-encrypted-cluster | passed |
| affinity | passed |
| self-healing-advanced | failed |
| one-pod | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| auto-tuning | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| security-context | passed |
| proxysql-sidecar-res-limits | passed |
| scheduled-backup | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | failed |
| scaling | passed |
| storage | passed |
| recreate | failed |
| init-deploy | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | failed |
| self-healing | passed |
| upgrade | passed |
| limits | passed |
| demand-backup | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| security-context | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| scheduled-backup | failed |
| one-pod | passed |
| auto-tuning | passed |
| demand-backup-encrypted-with-tls | failed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| big-data | failed |
| security-context | passed |
| recreate | failed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| init-deploy | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| restore-to-encrypted-cluster | failed |
| demand-backup | passed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| storage | passed |
| recreate | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| big-data | failed |
| operator-self-healing | passed |
| init-deploy | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| restore-to-encrypted-cluster | failed |
| demand-backup | passed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| big-data | passed |
| scaling | passed |
| recreate | passed |
| self-healing | passed |
| limits | passed |
| scaling-proxysql | passed |
| self-healing-advanced | passed |
| restore-to-encrypted-cluster | failed |
| monitoring | passed |
| upgrade | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | failed |

File: e2e-tests/security-context/conf/sec-context-workload-cluster-role.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (delgod) [inline e2e-tests/security-context/conf/sec-context-workload-cluster-role.yml:1]: is this file needed? I don't see where we apply it

File: e2e-tests/security-context/compare/cronjob_each-hour-pvc-oc.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
+apiVersion: batch/v1beta1
```
Comment (delgod) [inline e2e-tests/security-context/compare/cronjob_each-hour-pvc-oc.yml:1]: this file is identical to `e2e-tests/security-context/compare/cronjob_each-hour-pvc.yml` so I think it is not needed

File: e2e-tests/security-context/compare/statefulset_sec-context-proxysql-oc.yml [hunk 3]
```diff
@@ -0,0 +1,170 @@
+apiVersion: apps/v1
```
Comment (delgod) [inline e2e-tests/security-context/compare/statefulset_sec-context-proxysql-oc.yml:1]: this file is identical to `e2e-tests/security-context/compare/statefulset_sec-context-proxysql.yml` so I think it is not needed

File: e2e-tests/functions [hunk 4]
```diff
@@ -221,8 +221,8 @@ compare_kubectl() {
     fi
 
     kubectl_bin get -o yaml ${resource} \
-        | egrep -v "namespace:|uid:|resourceVersion:|selfLink:|creationTimestamp:|.*: default-token-.*|deletionTimestamp:|image:|clusterIP:|dataSource:|procMount:" \
-        | egrep -v "^  storageClassName:|finalizers:|kubernetes.io/pvc-protection|volumeName:|storage-provisioner:|status: \{\}|volumeMode: Filesystem" \
+        | egrep -v "namespace:|uid:|resourceVersion:|selfLink:|creationTimestamp:|.*: default-token-.*|- name: default-dockercfg-.*|deletionTimestamp:|image:|clusterIP:|dataSource:|procMount:" \
+        | egrep -v "^  storageClassName:|finalizers:|kubernetes.io/pvc-protection|volumeName:|storage-provisioner:|status: \{\}|volumeMode: Filesystem|imagePullSecrets:" \
```
[the remark this answers is not in the document]
Comment (cap1984) [reply e2e-tests/functions:226]: Unfortunately Openshift adds to the place which K8S does not change. I didn't want to, but need to.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| recreate | passed |
| big-data | passed |
| storage | passed |
| init-deploy | passed |
| self-healing | failed |
| scaling-proxysql | passed |
| limits | passed |
| self-healing-advanced | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | failed |
| monitoring | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| demand-backup | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | failed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |
| upgrade-consistency | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| recreate | failed |
| big-data | passed |
| scaling | passed |
| restore-to-encrypted-cluster | passed |
| limits | failed |
| monitoring | failed |
| scaling-proxysql | passed |
| self-healing | failed |
| monitoring-2-0 | passed |
| upgrade | failed |
| self-healing-advanced | failed |
| demand-backup | failed |
| scheduled-backup | failed |
| affinity | passed |
| operator-self-healing | passed |
| one-pod | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | passed |
| upgrade-consistency | passed |
| security-context | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| recreate | failed |
| init-deploy | passed |
| limits | failed |
| storage | passed |
| self-healing | failed |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | failed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| self-healing-advanced | failed |
| monitoring | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| scheduled-backup | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | passed |
| upgrade-consistency | passed |
| security-context | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| big-data | passed |
| scaling-proxysql | passed |
| upgrade | failed |
| recreate | passed |
| storage | passed |
| self-healing | failed |
| self-healing-advanced | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-encrypted-with-tls | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| security-context | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | failed |
| storage | passed |
| self-healing | failed |
| big-data | passed |
| init-deploy | passed |
| recreate | passed |
| limits | passed |
| restore-to-encrypted-cluster | passed |
| monitoring | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| scheduled-backup | passed |
| affinity | passed |
| demand-backup-encrypted-with-tls | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| upgrade-consistency | passed |
| security-context | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| init-deploy | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| limits | passed |
| big-data | passed |
| storage | passed |
| demand-backup-encrypted-with-tls | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| self-healing | failed |
| scaling | passed |
| affinity | passed |
| scaling-proxysql | passed |
| one-pod | passed |
| upgrade | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| security-context | passed |
